### DISCUSSION TITLE
Increasing the "display length limit" in results

### DISCUSSION THREAD

**Discussion by intrigus-lgtm** [github/securitylab#53] [General] [answered]
Hi,
when I do a simple query like:
````
import javascriptfrom Property p
select p.getInit()
````
I receive :
`PUBLIC_ ... MISSION`

instead of the full expression:
`PUBLIC_ROUTE_PERMISSION`

Is there a way to increase this “display length limit”?

(Similar question originally asked by Melvin on Slack)

**Comment by asgerf** [comment, marked as answer]
Unfortunately not. It's not a display setting, it's a limitation in how the string representation of an expression is constructed in our JavaScript libraries. We're aware that this is an issue and have plans to improve it.

As a workaround you can do:
```codeql
import javascript

from Property p
select describeExpression(p.getInit())
```

edit: fixed typo in predicate name

**Comment by pieterhartel** [comment]
The suggested work around no longer works. What is the current solution for this issue?

**Reply by asgerf** [reply]
Looks like I misspelled the predicate name in my original answer, it's actually called `describeExpression`, not `describeExpr`.

```ql
import javascript

from Property p
select describeExpression(p.getInit())
```
